Generals/Code/GameEngine/Source/GameClient/MessageStream/SelectionXlat.cpp

```diff
@@ -1097,57 +1097,47 @@ GameMessageDisposition SelectionTranslator::translateGameMessage(const GameMessa
 					m_lastGroupSelTime = now;
 				}
 
-				// check for double-press to jump view
+				Drawable* draw = TheInGameUI->getFirstSelectedDrawable();
+				if (draw && draw->isKindOf(KINDOF_STRUCTURE))
+				{
+					//Kris: Jan 12, 2005
+					//Can't select other units if you have a structure selected. So deselect the structure to prevent
+					//group force attack exploit.
+					TheInGameUI->deselectAllDrawables();
+				}
 
-				if ( now - m_lastGroupSelTime < 20 && group == m_lastGroupSelGroup )
+				TheInGameUI->deselectAllDrawables(false); //No need to post message because we're just creating a new group!
+
+				Player* player = ThePlayerList->getLocalPlayer();
+				if (player)
 				{
-					DEBUG_LOG(("META: DOUBLETAP select team %d",group));
-					Player *player = ThePlayerList->getLocalPlayer();
-					if (player)
+					Squad* selectedSquad = player->getHotkeySquad(group);
+					if (selectedSquad != NULL)
 					{
-						Squad *selectedSquad = player->getHotkeySquad(group);
-						if (selectedSquad != NULL)
+						VecObjectPtr objlist = selectedSquad->getLiveObjects();
+						Int numObjs = objlist.size();
+						for (Int i = 0; i < numObjs; ++i)
 						{
-							VecObjectPtr objlist = selectedSquad->getLiveObjects();
-							Int numObjs = objlist.size();
-							if (numObjs > 0)
+							if (objlist[i]->getControllingPlayer() == player)
 							{
-								// if theres someone in the group, center the camera on them.
-								TheTacticalView->lookAt( objlist[numObjs-1]->getDrawable()->getPosition() );
+								TheInGameUI->selectDrawable(objlist[i]->getDrawable());
 							}
 						}
-					}
-
-				}
-				else
-				{
-
-					Drawable *draw = TheInGameUI->getFirstSelectedDrawable();
-					if( draw && draw->isKindOf( KINDOF_STRUCTURE ) )
-					{
-						//Kris: Jan 12, 2005
-						//Can't select other units if you have a structure selected. So deselect the structure to prevent
-						//group force attack exploit.
-						TheInGameUI->deselectAllDrawables();
-					}
 
-					// no need to send two messages for selecting the same group.
-					TheMessageStream->appendMessage((GameMessage::Type)(GameMessage::MSG_ADD_TEAM0 + group));
-					Player *player = ThePlayerList->getLocalPlayer();
-					if (player)
-					{
-						Squad *selectedSquad = player->getHotkeySquad(group);
-						if (selectedSquad != NULL)
+						// check for double-press to jump view
```

Comment:
I see you fixed it in Generals for `MSG_META_ADD_TEAM` and in Zero Hour for `MSG_META_SELECT_TEAM`. Something is not right.

Does the same issue practically also apply to `MSG_META_ADD_TEAM`? Do both need fixing?